votor/src/certificate_pool.rs

```diff
@@ -246,15 +246,16 @@ impl CertificatePool {
                 continue;
             }
             let mut vote_certificate_builder = VoteCertificateBuilder::new(cert_id);
-            vote_types.iter().for_each(|vote_type| {
+            vote_types.iter().enumerate().for_each(|(vote_type_index, vote_type)| {
+                let is_first_vote_type = vote_type_index == 0;
                 if let Some(vote_pool) = self.vote_pools.get(&(slot, *vote_type)) {
-                match vote_pool {
-                    VotePoolType::SimpleVotePool(pool) => pool.add_to_certificate(&mut vote_certificate_builder),
-                    VotePoolType::DuplicateBlockVotePool(pool) => pool.add_to_certificate(block_id.as_ref().expect("Duplicate block pool for {vote_type:?} expects a block id for certificate {cert_id:?}"), &mut vote_certificate_builder),
-                };
-            }
+                    match vote_pool {
+                        VotePoolType::SimpleVotePool(pool) => pool.add_to_certificate(is_first_vote_type, &mut vote_certificate_builder),
+                        VotePoolType::DuplicateBlockVotePool(pool) => pool.add_to_certificate(block_id.as_ref().expect("Duplicate block pool for {vote_type:?} expects a block id for certificate {cert_id:?}"), is_first_vote_type, &mut vote_certificate_builder),
+                    };
```

Comment:
Passing along this `is_first_vote_type` seems a bit awkward and error prone, because it relies on the ordering returned by `certificate_limits_and_vote_types`

1. Because `VoteCertificateBuilder` already tracks the type of certificate it is building, `VoteCertificateBuilder::aggregate()` should be able to deduce if passed a `VoteType` whether to add to bitmap1 or bitmap2
2. The vote pool `add_to_certificate()` method should be able to just pass its internal `VoteType` to the `vote_certificate_builder`

Reply:
Okay, I reverted to using VoteType now.